File: asion-base/src/test/java/org/asion/demo/domain/model/AccountRepository.java
package org.asion.demo.domain.model;

import org.asion.base.ddd.domain.BaseRepository;
import org.asion.base.ddd.domain.annotations.DomainRepository;

/**
 * @author Asion.
 * @since 2017/3/20.
 */
@DomainRepository
public interface AccountRepository extends BaseRepository<Long, Account> {
}

File: asion-account/asion-account-core/src/main/java/io/vincent/account/domain/model/AccountRepository.kt
package io.vincent.account.domain.model

import io.vincent.common.vo.Page
import org.asion.base.ddd.domain.BaseRepository
import org.asion.base.ddd.domain.annotations.DomainRepository

/**
 * @author Vincent.
 * @since 16/4/30.
 */
@DomainRepository
interface AccountRepository : BaseRepository<Account, Long> {
    fun findByUsername(username: String): Account?
    fun findPage(pageNumber: Int, pageSize: Int): Page<Account>
    fun updateStatus(id: Long, statusCode: Int): Long?
}
